pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -0,0 +1,374 @@
+package vsphereprivate
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	internal "github.com/openshift/installer/pkg/terraform/exec/plugins/vsphereprivate/internal"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/nfc"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/ovf"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func resourceVSpherePrivateImportOva() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceVSpherePrivateImportOvaCreate,
+		Read:          resourceVSpherePrivateImportOvaRead,
+		Update:        resourceVSpherePrivateImportOvaUpdate,
+		Delete:        resourceVSpherePrivateImportOvaDelete,
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "",
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"filename": {
+				Type:         schema.TypeString,
+				Description:  "",
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"datacenter": {
+				Type:        schema.TypeString,
+				Description: "The ID of the datacenter. Can be ignored if creating a datacenter folder, otherwise required.",
+				Required:    true,
+			},
+			"cluster": {
+				Type:        schema.TypeString,
+				Description: "The ID of the datacenter. Can be ignored if creating a datacenter folder, otherwise required.",
+				Required:    true,
+			},
+			"network": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of a network that the virtual machine will use.",
+			},
+			"datastore": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the virtual machine's datastore. The virtual machine configuration is placed here, along with any virtual disks that are created without datastores.",
+			},
+			"folder": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the folder to locate the virtual machine in.",
+			},
+		},
+	}
+}
+
+// ImportOvaParams contains the vCenter objects required to import a OVA into vSphere.
+type ImportOvaParams struct {
+	ResourcePool *object.ResourcePool
+	Datacenter   *object.Datacenter
+	Datastore    *object.Datastore
+	Network      *object.Network
+	Host         *object.HostSystem
+	Folder       *object.Folder
+}
+
+func findImportOvaParams(client *vim25.Client, datacenter, cluster, datastore, network string) (*ImportOvaParams, error) {
```

Comment:
there is a mix of context.TODO and ctx being passed to various functions in this functions...?

also why not use the defaultAPITimeout to create a context with a deadline?